Index: libsdl1.2-1.2.11/configure.in
===================================================================
--- libsdl1.2-1.2.11.orig/configure.in	2006-06-27 06:48:33.000000000 +0200
+++ libsdl1.2-1.2.11/configure.in	2006-07-19 14:51:37.000000000 +0200
@@ -513,10 +513,10 @@
     if test x$enable_audio = xyes -a x$enable_nas = xyes; then
         AC_MSG_CHECKING(for NAS audio support)
         have_nas=no
-        if test -r /usr/X11R6/include/audio/audiolib.h; then
+        if test -r /usr/include/audio/audiolib.h; then
             have_nas=yes
-            NAS_CFLAGS="-I/usr/X11R6/include/"
-            NAS_LIBS="-L/usr/X11R6/lib -laudio -lXt"
+            NAS_CFLAGS="-I/usr/include/"
+            NAS_LIBS="-L/usr/lib -laudio -lXt"
 
         dnl On IRIX, the NAS includes are in a different directory,
         dnl and libnas must be explicitly linked in
