Index: libsdl1.2-1.2.11/sdl.pc.in
===================================================================
--- libsdl1.2-1.2.11.orig/sdl.pc.in	2006-07-20 14:14:40.000000000 +0200
+++ libsdl1.2-1.2.11/sdl.pc.in	2006-07-20 14:14:47.000000000 +0200
@@ -10,5 +10,5 @@
 Version: @SDL_VERSION@
 Requires:
 Conflicts:
-Libs: -L${libdir} @SDL_RLD_FLAGS@ @SDL_LIBS@
+Libs: -L${libdir} @SDL_RLD_FLAGS@ -lSDL
 Cflags: -I${includedir}/SDL @SDL_CFLAGS@
Index: libsdl1.2-1.2.11/sdl-config.in
===================================================================
--- libsdl1.2-1.2.11.orig/sdl-config.in	2006-07-20 14:16:00.000000000 +0200
+++ libsdl1.2-1.2.11/sdl-config.in	2006-07-20 14:16:20.000000000 +0200
@@ -44,7 +44,7 @@
       echo -I@includedir@/SDL @SDL_CFLAGS@
       ;;
 @ENABLE_SHARED_TRUE@    --libs)
-@ENABLE_SHARED_TRUE@      echo -L@libdir@ @SDL_RLD_FLAGS@ @SDL_LIBS@
+@ENABLE_SHARED_TRUE@      echo -L@libdir@ @SDL_RLD_FLAGS@ -lSDL
 @ENABLE_SHARED_TRUE@      ;;
 @ENABLE_STATIC_TRUE@@ENABLE_SHARED_TRUE@    --static-libs)
 @ENABLE_STATIC_TRUE@@ENABLE_SHARED_FALSE@    --libs|--static-libs)
